src/components/sync-connected-sites.tsx

```diff
@@ -54,13 +63,41 @@ export const convertTreeToOptionsToSync = ( tree: TreeNode[] ): SyncOption[] =>
 		const wpContent = filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
 
 		wpContent.forEach( ( item ) => {
-			if ( item.checked && isSyncOption( item.id ) ) {
-				optionsToSync.push( item.id );
+			if ( isSyncOption( item.id ) ) {
+				const hasCheckedChildren = item.children?.some( ( child ) => child.checked );
+				const shouldInclude = item.checked || hasCheckedChildren;
+
+				if ( shouldInclude ) {
```

Comment:
If parent is checked, then it means that at least one child is checked, and if it's not true, then it means that the tree has some bug and works incorrectly.
So, it's better to keep code more simple and at the same time don't mute bugs with tree state.
Does it make sense or am I missing something?
```suggestion
				if ( item.checked ) {
```

Reply:
I don't think that is correct. The parent is `checked` only if all children are selected.
This change handles the scenario where we have some children checked but not all. In this case we still want to include `plugins` for example, if only of the plugins was selected.